### PR TITLE
docs: Add the SAML intro docs

### PR DESCRIPTION
**Pull request by dakshitha**
Please refer to this issue for more information: https://github.com/zitadel/zitadel/issues/7436

### CODE REVIEW

**Comment by dakshitha** [comment]
@stebenz I have added you as a reviewer too since @livio-a is at the conference.

File: docs/docs/guides/integrate/login/saml.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+```
+
+In this SAML request:
+
+- The **ID** (id-8LjuzBEUQFYFWjL55) is generated by the service provider, and the identity provider will respond to this ID in the SAML response.
+- **IssueInstant** (2024-05-11T04:13:52Z) is the timestamp for this request, used by the identity provider for verification to ensure the request is within an acceptable time frame.
+- **Destination** (https://my-instance-xtzfbc.zitadel.cloud/saml/v2/SSO) points to the identity provider's URL, ensuring the request is sent to the correct recipient.
+- **AssertionConsumerServiceURL** (http://127.0.0.1:5000/acs) indicates where the response should be sent after user authentication.
+- **Issuer** (https://zitadel-test.sp/metadata) is a predefined string formatted as a URL, matching what the identity provider expects.
+- **Signature** ensures the integrity and authenticity of the SAML request. It includes:
```
Comment (stebenz) [inline docs/docs/guides/integrate/login/saml.md:115]: I would add a space or something to have a split as SignedInfo, Reference, SignatureValue and KeyInfo are inside Signature

File: docs/docs/guides/integrate/login/saml.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+- **SignedInfo** contains details about the canonicalization and signature methods.
+- **Reference** points to the signed data and includes a transform and digest method.
+- **SignatureValue** provides the actual digital signature.
+- **KeyInfo** containing the X.509 certificate, which holds the public key used to verify the signature.
+
+These XMLs are stringified, encoded, and passed in the query string to the identity provider, where the user is authenticated, and the SAML response is prepared, which will be received by the SP as an encoded string.
```
Comment (stebenz) [inline docs/docs/guides/integrate/login/saml.md:121]: What do you mean as query string? If you mean as URL parameters, then maybe a comment to the different Bindings would be necessary as well, as the request is send as parameter or request body related to the bindings.

Comment (stebenz) [reply docs/docs/guides/integrate/login/saml.md:121]: It's also dependend on the binding, as to how the signature is sent.

File: docs/docs/guides/integrate/login/saml.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+				<AttributeValue>260242264868201995</AttributeValue>
+			</Attribute>
+		</AttributeStatement>
+	</Assertion>
+</Response>
+```
+
+In this SAML response:
+
+- The **ID** (_164ba12b-6711-40e0-8ddb-55aa810f1c92) matches the ID from the SAML request (id-8LjuzBEUQFYFWjL55), allowing the service provider to verify it.
```
Comment (stebenz) [inline docs/docs/guides/integrate/login/saml.md:202]: Which part of the content of this 2 ID's match here?

File: docs/docs/guides/integrate/login/saml.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+- **InResponseTo** references the ID of the SAML request.
+- **IssueInstant** (2024-06-11T04:17:41Z) is the timestamp of the response.
+- **Destination** (http://127.0.0.1:5000/acs) confirms that the response is intended for the service provider's Assertion Consumer Service URL.
+- **Issuer** (https://my-instance-xtzfbc.zitadel.cloud/saml/v2/metadata) is the identity provider's unique string, used to verify the response's origin.
+- **StatusCode** indicates the status of the authentication process, with a value of Success.
```
Comment (stebenz) [inline docs/docs/guides/integrate/login/saml.md:207]: Mabye a comment that if not Successfull there was a problem with the login?

File: docs/docs/guides/integrate/login/saml.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+- **Signature** ensures the integrity and authenticity of the SAML response. It includes:
```
Comment (stebenz) [inline docs/docs/guides/integrate/login/saml.md:208]: Same comment about the spacing as above.

File: docs/docs/guides/integrate/login/saml.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+- **AttributeStatement** contains additional user attributes.
+
+## SAML identity brokering
+
+### How SAML identity brokering works
+
+- **Initial Authentication Request**: A user attempts to access a service (SP1) that is protected by an IdP (IdP1).
+- **Redirection to IdP1**: The user is redirected to IdP1 for authentication. If IdP1 trusts another IdP (IdP2) for authentication, it will redirect the user to IdP2.
+- **Authentication at IdP2**: The user authenticates with IdP2, which generates a SAML assertion containing the user's identity and attributes.
+- **Assertion Transfer**: IdP2 sends the SAML assertion back to IdP1. IdP1 processes this assertion and may add additional attributes or modify the assertion based on its own policies.
```
Comment (stebenz) [inline docs/docs/guides/integrate/login/saml.md:226]: There should be no transfer? If ZITADEL is IdP1, the Assertion from IdP2 is processed and an independent assertion is created and used as response to SP1.

File: docs/docs/guides/integrate/login/saml.md (hunk 1 above)
Comment (dakshitha) [reply docs/docs/guides/integrate/login/saml.md:115]: Done

File: docs/docs/guides/integrate/login/saml.md (hunk 5 above)
Comment (dakshitha) [reply docs/docs/guides/integrate/login/saml.md:208]: Done

File: docs/docs/guides/integrate/login/saml.md (hunk 2 above)
Comment (dakshitha) [reply docs/docs/guides/integrate/login/saml.md:121]:
```suggestion
These XMLs are stringified, encoded, and transmitted according to the SAML binding used. For instance, in HTTP Redirect Binding, the request is sent as URL parameters, while in HTTP POST Binding, it is included in the request body. The signature's transmission method also depends on the binding used, ensuring the integrity and authenticity of the SAML message across different communication channels.
```

File: docs/docs/guides/integrate/login/saml.md (hunk 3 above)
Comment (dakshitha) [reply docs/docs/guides/integrate/login/saml.md:202]: ID="_164ba12b-6711-40e0-8ddb-55aa810f1c92" InResponseTo="id-8LjuzBEUQFYFWjL55"

File: docs/docs/guides/integrate/login/saml.md (hunk 4 above)
Comment (dakshitha) [reply docs/docs/guides/integrate/login/saml.md:207]:
```suggestion
- **StatusCode** indicates the status of the authentication process, with a value of "Success." If the status is not "Success," it means there was a problem with the login, such as incorrect credentials or other authentication issues.
```

File: docs/docs/guides/integrate/login/saml.md (hunk 6 above)
Comment (dakshitha) [reply docs/docs/guides/integrate/login/saml.md:226]:
```suggestion
- **Assertion Processing**: ZITADEL (IdP1) processes the SAML assertion from IdP2. ZITADEL then creates an independent SAML assertion based on the information received and its own policies.
- **Response to SP1**: ZITADEL (IdP1) sends this newly created SAML assertion to SP1, completing the authentication process.
```

**Review by stebenz — APPROVED** [review]

**Pull request merged**